File: homeworks/hw1/q5.cpp
class Solution {
public:
    int maxSubarraySumCircular(vector<int>& nums) {
        int curMax = 0,maxSum = nums[0],curMin = 0,minSum = nums[0], total = 0;
        for(const int& n : nums){
            curMax = max(curMax+n,n);
            maxSum = max(maxSum,curMax);
            curMin = min(curMin+n,n);
            minSum = min(minSum,curMin);
            total += n;
        }
        return (maxSum > 0) ? max(maxSum,total-minSum) : maxSum; 
    }
};

File: homeworks/hw4/q5.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        i,mr,last,jumps=0,0,0,0
        while(last < n-1):
            mr = max(mr,i + nums[i])
            if i == last: 
                last = mr
                jumps += 1
            i += 1
        return jumps
